### ISSUE TITLE
Inserting into SizedCache can cause the cache size to decrease

### ISSUE THREAD

**Issue by randombit** [jaemk/cached#137]
Apologies if this is documented but I could not find anything describing this behavior, and at least to me it seems very unexpected.

Using `cache_set` to insert an element that already exists in a `SizedCache` can cause the overall size of the cache to decrease.  I would expect that once `SizedCache` has reached its maximum configured size, it would stay that way perpetually.

```
use cached::{Cached, SizedCache};

fn main() {
    let mut cache = SizedCache::<u32, ()>::with_size(3);

    cache.cache_set(1, ());
    cache.cache_set(2, ());
    cache.cache_set(3, ());

    assert!(cache.cache_get(&1).is_some());
    assert!(cache.cache_get(&2).is_some());
    assert!(cache.cache_get(&3).is_some());
    assert!(!cache.cache_get(&4).is_some());

    cache.cache_set(4, ());
    assert_eq!(cache.cache_size(), 3);

    cache.cache_set(4, ());
    assert_eq!(cache.cache_size(), 3); // fails! actually 2 now

    assert!(!cache.cache_get(&1).is_some());
    assert!(cache.cache_get(&2).is_some()); // fails! 2 is also evicted
    assert!(cache.cache_get(&3).is_some());
    assert!(cache.cache_get(&4).is_some());
}
```

**Comment by jaemk** [comment]
Oh boy, that's a bug. Looks like we're missing a call to `remove_index` after this line https://github.com/jaemk/cached/blob/f5911dc3fbc03e1db9f87192eb854fac2ee6ac98/src/stores/sized.rs#L184
That ends up resulting in the `store` size remaining "3" even after the key "1" gets evicted by your first `cache.cache_set(4, ());`

**Comment by jaemk** [comment]
fix is building in #138

**Comment by jaemk** [comment]
fix released in 0.41.0

**Comment by randombit** [comment]
Thank you for the fast fix!
